src/main/java/com/joss/conductor/mobile/util/ScreenShotUtil.java

```diff
@@ -14,33 +15,55 @@
 
     private static final String DIR = "/target/test-screenshots";
     private static final String WORKING_DIR = System.getProperty("user.dir");
+    private static final String PNG_EXT = ".png";
 
     public static void take(Locomotive locomotive, String testName) {
-        try {
-            FileUtils.copyFile(locomotive.driver.getScreenshotAs(OutputType.FILE),
-                    new File(WORKING_DIR
-                            + File.separator
-                            + DIR
-                            + File.separator
-                            + testName.replace(File.separator, "-") + ".png"));
-        } catch (IOException e) {
-            e.printStackTrace();
-        }
+        writeFile(locomotive.driver, createFilePathAndName(testName));
     }
 
     public static void take(Locomotive locomotive, String path, String testName) {
+        writeFile(locomotive.driver, createFilePathAndName(path, testName));
+    }
+
+    private static void writeFile(AppiumDriver appiumDriver, String filePathAndName) {
         try {
-            FileUtils.copyFile(locomotive.driver.getScreenshotAs(OutputType.FILE),
-                    new File(WORKING_DIR
-                            + File.separator
-                            + DIR
-                            + File.separator
-                            + path.replace(File.separator, "-")
-                            + File.separator
-                            + testName.replace(File.separator, "-") + ".png"));
+            FileUtils.copyFile(appiumDriver.getScreenshotAs(OutputType.FILE), new File(filePathAndName));
         } catch (IOException e) {
             e.printStackTrace();
         }
     }
+
+    private static String createFilePathAndName(String testName) {
+        return WORKING_DIR
+                + File.separator
+                + DIR
+                + File.separator
+                + limitLengthTo100Chars(removeInvalidFilenameChars(testName))
+                + PNG_EXT;
+    }
+
+    private static String createFilePathAndName(String path, String testName) {
+        return WORKING_DIR
+                + File.separator
+                + DIR
+                + File.separator
+                + limitLengthTo100Chars(removeInvalidFilenameChars(path))
+                + File.separator
+                + limitLengthTo100Chars(removeInvalidFilenameChars(testName))
+                + PNG_EXT;
+    }
+
+    private static String removeInvalidFilenameChars(String name) {
+        return name.replace(File.separator, "-")
```

Comment:
The only thing to consider here is that if you have "thisTest*" and "thisTest?" it'll end up being the same files. If that's common enough you might want to add a timestamp to the end of the filename